multivariate_single_forecast.py

```diff
@@ -1,128 +1,119 @@
-features = e.data[features_considered]
-features.index = e.data.index
-# assert(list(features)[0] == 'Close')
-features = features.dropna()
-features = features[26:]
-
-
-# features.plot(subplots=True)
-# plt.show()
-
-dataset = features.values
-x_train_single, y_train_single, x_val_single, y_val_single = split_multivariate(dataset, history_size, target_distance,
-                                                                                step, single_step=True, classification=CLASSIFICATION)
-
-print('Single window of past history : {}'.format(x_train_single[0].shape))
-
-train_data_single = tf.data.Dataset.from_tensor_slices(
-    (x_train_single, y_train_single))
-train_data_single = train_data_single.cache().shuffle(
-    BUFFER_SIZE).batch(BATCH_SIZE).repeat()
-
-val_data_single = tf.data.Dataset.from_tensor_slices(
-    (x_val_single, y_val_single))
-val_data_single = val_data_single.batch(BATCH_SIZE).repeat()
-
-single_step_model = tf.keras.models.Sequential()
-
-single_step_model.add(tf.keras.layers.LSTM(32, return_sequences=True,
-                                           input_shape=x_train_single.shape[-2:]))
-# single_step_model.add(tf.keras.layers.LSTM(32, input_shape=x_train_single.shape[-2:]))
-
-single_step_model.add(tf.keras.layers.LSTM(16, activation='relu'))
-if CLASSIFICATION:
-  single_step_model.add(tf.keras.layers.Dense(16, activation='relu'))
-  single_step_model.add(tf.keras.layers.Dense(1, activation='sigmoid'))
-else:
-  single_step_model.add(tf.keras.layers.Dense(1))
-
-# single_step_model.compile(optimizer=tf.keras.optimizers.RMSprop(), loss='mae')
-single_step_model.compile(optimizer=tf.keras.optimizers.Adam(), loss=tf.keras.losses.BinaryCrossentropy(), metrics=['accuracy'])
-
-val_callback = tf.keras.callbacks.ModelCheckpoint(
-    'checkpoints/multivariate_single_model', monitor='val_accuracy', verbose=1, save_best_only=True,
-    save_weights_only=False, mode='auto', save_freq='epoch'
-)
-
-single_step_history = single_step_model.fit(train_data_single, epochs=EPOCHS,
-                                            steps_per_epoch=EVALUATION_INTERVAL,
-                                            validation_data=val_data_single,
-                                            validation_steps=50, callbacks=[val_callback])
-
-
-plot_train_history(single_step_history,
-                   'Single Step Training and validation loss')
-
-
-for x, y in val_data_single.take(10):
-  y_pred = single_step_model.predict(x)[0]
-  print(f"prediction: {y_pred}")
-  if CLASSIFICATION:
-    if y_pred >= 0.5:
-      y_pred = 1
-    else:
-      y_pred = 0
-  plot = show_plot([x[0][:, 0].numpy(), y[0].numpy(),
-                    y_pred], target_distance,
-                   'Single Step Prediction')
-  plot.show()
+def get_lstm():
+  """
+  Keras LSTM Architecture 
+  """
+  ssm = tf.keras.models.Sequential()
+
+  ssm.add(tf.keras.layers.LSTM(64, return_sequences=True,
+                                input_shape=xt.shape[-2:]))
```

Comment:
what is xt?

Reply:
xtrain

Reply:
oh okay yea, ill fix that rn

Reply:
so I guess this is the n_dimensions? based off input dimension, like OHLC would be 4 dimensions?

OHLC + MA = 5 dim? right?

Reply:
so maybe this should be set to `len(FEATURES)`?

Reply:
I don't think so, that's the 3d tensor that you input in keras. your old code is doing the same ?

![image](https://user-images.githubusercontent.com/12208440/87476257-ee503e00-c5da-11ea-81bb-b6e0f7886881.png)

Reply:
yeah so we have `(batch_size, time_steps, number_of_features)`

for input shape I think we drop batch_size, so `(time_steps, number_of_features)`, so I think its `(HISTORY_SIZE, len(FEATURES))`

can you print what `xt[-2:]` was and confirm?

Reply:
Id rather have this value be coded and not set dynamically by dataset

Reply:
I see what ur saying, will work on it later

Reply:
input makes sense, in the process of changing it to static

Reply:
got it